core/base-service/base.js

```diff
@@ -95,47 +97,16 @@ module.exports = class BaseService {
    * Route to mount this service on
    *
    * @abstract
-   * @return {object} route
-   * @return {string} route.base
-   *    (Optional) The base path of the routes for this service.
-   *    This is used as a prefix.
-   * @return {string} route.pattern
-   *    A path-to-regexp pattern defining the route pattern and param names
-   *    See https://www.npmjs.com/package/path-to-regexp
-   * @return {Regexp} route.format
-   *    Deprecated: Regular expression to use for routes for this service's badges
-   *    Use `pattern` instead
-   * @return {string[]} route.capture
-   *    Deprecated: Array of names for the capture groups in the regular
-   *    expression. The handler will be passed an object containing
-   *    the matches.
-   *    Use `pattern` instead
-   * @return {Joi.object} route.queryParamSchema
-   *    (Optional) A Joi schema (`Joi.object({ ... }).required()`)
-   *    for the query param object. If you know a parameter
-   *    will never receive a numeric string, you can use
-   *    `Joi.string()`. Because of quirks in Scoutcamp and Joi,
-   *    alphanumeric strings should be declared using
-   *    `Joi.alternatives().try(Joi.string(), Joi.number())`,
-   *    otherwise a value like `?success_color=999` will fail.
-   *    A parameter requiring a numeric string can use
-   *    `Joi.number()`. A parameter that receives only non-numeric
-   *    strings can use `Joi.string()`. A parameter that never
-   *    receives numeric can use `Joi.string()`. A boolean
-   *    parameter should use `Joi.equal('')` and will receive an
-   *    empty string on e.g. `?compact_message` and undefined
-   *    when the parameter is absent. (Note that in,
-   *    `examples.queryParams` boolean query params should be given
-   *    `null` values.)
+   * @type {module:core/base-service/base~Route}
```

Comment:
We have to use this syntax in quite a few places because of this https://github.com/jsdoc/jsdoc/issues/969

Its fine.. until we move stuff about, at which point we'll probably forget and break links. Hopefully, that is where the `jsdoc/no-undefined-types` lint rule steps in to save us :crossed_fingers:

Reply:
That makes sense and good that it will (probably/hopefully) get caught in the linting process. 

This is another one of those times when I'd hope danger could do some automagic to detect but that seems like it would be pretty challenging to check precisely. _Maybe_  danger could detect file renames/directory changes under `./core` and print a warning like we do for [serverSecrets](https://github.com/badges/shields/blob/master/dangerfile.js#L115)?